libutee: fix TEE_MemMove()'s return type

vincent-mailhol opened the pull request:
In all the published versions of the TEE Internal API Specification, the return type of `TEE_MemMove()` has always been `void`, not `void *`.

jenswi-linaro commented on lib/libutee/tee_api.c:
```diff
@@ -607,14 +607,14 @@ const void *TEE_GetInstanceData(void)
+void TEE_MemMove(void *dest, const void *src, size_t size)
 {
-	return memmove(dest, src, size);
+	(void)memmove(dest, src, size);
```
Please drop the cast, same below.

vincent-mailhol replied:
No problem. I removed the cast in https://github.com/OP-TEE/optee_os/pull/7140/commits/312fd9960ca51a8172adad2fdcef218c97ca5ae7.

vincent-mailhol replied:
Hug?! I did not know that fixup commits also needed to be signed. Regardless, fixed again in https://github.com/OP-TEE/optee_os/pull/7140/commits/311d28d1320f385ffefcc15736f2e669ecfab0ac.

etienne-lms reviewed (commented):
Could you update the commit message header line: prefix with `libutee: `, e.g.:
`libutee: fix TEE_MemMove()'s return type`?

`Reviewed-by: Etienne Carriere <etienne.carriere@foss.st.com>`

By the way, nice `Fixes:` reference :-)

vincent-mailhol commented:
> Could you update the commit message header line: prefix with `libutee: `, e.g.: `libutee: fix TEE_MemMove()'s return type`?
> 
> `Reviewed-by: Etienne Carriere <etienne.carriere@foss.st.com>`

I squashed the fixup, changed the commit title and added your review tag in https://github.com/OP-TEE/optee_os/pull/7140/commits/b05a0983b5be3414280d378244fae480e9c62d91.

> By the way, nice `Fixes:` reference :-)

Thank you ^^

jenswi-linaro commented:
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`
